File: playwright-python-tests/page-objects/IssuesPage.ts
import { Page, Locator } from '@playwright/test';

export class IssuesPage {
  readonly page: Page;
  readonly titleField: Locator;
  readonly descField: Locator;
  readonly submitIssueBtn: Locator;
  readonly validationWarning: Locator;

  constructor(page: Page) {
    this.page = page;
    this.titleField = page.locator('#issue_title');
    this.descField = page.locator('#issue_body');
    this.submitIssueBtn = page.locator("button:has-text('Submit new issue')");
    this.validationWarning = page.locator('.blank-validation-error');
  }

  async submitIssue(title: string, description: string): Promise<void> {
    await this.titleField.fill(title);
    await this.descField.fill(description);
    await this.submitIssueBtn.click();
  }
}

File: pytest-python-tests/pages/issues_page.py
from .base_page import BasePage

class IssuesPage(BasePage):
    """IssuesPage represents custom GitHub Issues creation/editing form."""
    def __init__(self, page):
        super().__init__(page)
        self.title_field = "#issue_title"
        self.desc_field = "#issue_body"
        self.submit_issue_btn = "button:has-text('Submit new issue')"
        self.validation_warning = ".blank-validation-error"

    def submit_issue(self, title: str, description: str) -> None:
        self.fill_input(self.title_field, title)
        self.fill_input(self.desc_field, description)
        self.click(self.submit_issue_btn)
